default/Dockerfile

```diff
@@ -1,22 +1,17 @@
-FROM node:20-alpine AS development-dependencies-env
-COPY . /app
+FROM node:20-alpine AS base
+USER node
 WORKDIR /app
-RUN npm ci
-
-FROM node:20-alpine AS production-dependencies-env
 COPY ./package.json package-lock.json /app/
-WORKDIR /app
-RUN npm ci --omit=dev
 
-FROM node:20-alpine AS build-env
+FROM base AS staging
+RUN npm ci
 COPY . /app/
-COPY --from=development-dependencies-env /app/node_modules /app/node_modules
-WORKDIR /app
-RUN npm run build
+# Build app and prepare `node_modules` for production
+# This second install will remove the omitted deps automatically
+RUN npm run build \
+	&& npm ci --omit=dev
```

Comment:
Putting this in a the same stage as the previous one to reduce network, disk and time usage